src/main/java/org/apache/maven/shared/incremental/IncrementalBuildHelper.java

```diff
@@ -129,6 +146,30 @@ public void setDirectoryScanner( DirectoryScanner directoryScanner )
         this.directoryScanner = directoryScanner;
     }
 
+    /**
+     * Get the existing DirectoryScanner used by this helper,
+     * or create new a DirectoryScanner if none is yet set.
+     * The DirectoryScanner is used for detecting changes in a directory
+     */
+    public DirectoryScanner getGeneratedSourcesDirectoryScanner()
+    {
+        if ( generatedSourcesDirectoryScanner == null )
+        {
+            generatedSourcesDirectoryScanner = new DirectoryScanner();
+        }
+
+        return generatedSourcesDirectoryScanner;
+    }
+
+    /**
+     * Set the DirectoryScanner which shall get used by this build helper.
+     * @param generatedSourcesDirectoryScanner
+     */
+    public void setGeneratedSourcesDirectoryScanner( DirectoryScanner generatedSourcesDirectoryScanner )
```

Comment:
I'm not sure why these methods (or this class for that matter) is needed. It seems a directory scanner could be easily created without it.

Reply:
But I agree entirely with you that this class seems pointless. But I have no idea who uses it so it doesn't seem safe to change? Or would you prefer that I delete/inline this class.

Reply:
Though this class is literally the only thing in `maven-shared-incremental`.

Reply:
Don't change existing public methods in this PR, but also don't compound problems by introducing new ones we don't need.